smartapps/juano2310/jawbone-up-connect.src/jawbone-up-connect.groovy

```diff
@@ -28,16 +27,16 @@ mappings {
 	path("/receivedToken") { action: [ POST: "receivedToken", GET: "receivedToken"] }
 	path("/receiveToken") { action: [ POST: "receiveToken", GET: "receiveToken"] }
 	path("/hookCallback") { action: [ POST: "hookEventHandler", GET: "hookEventHandler"] }
+  path("/oauth/initialize") {action: [GET: "oauthInitUrl"]}
 	path("/oauth/callback") { action: [ GET: "callback" ] }
 }
 
-def getSmartThingsClientId() {
-   return appSettings.clientId
-}
-
-def getSmartThingsClientSecret() {
-   return appSettings.clientSecret
-}
+def getSmartThingsClientId() { return appSettings.clientId }
+def getSmartThingsClientSecret() { return appSettings.clientSecret }
+def getServerUrl() { return "https://graph.api.smartthings.com" }
+def getShardUrl() { return getApiServerUrl() }
+def getBuildRedirectUrl() { "${serverUrl}/oauth/initialize?appId=${app.id}&access_token=${state.accessToken}&apiServerUrl=${shardUrl}" }
+def buildRedirectUrl(page) { return buildActionUrl(page) }
```

Comment:
Why define a function that defers to another function?  Again a purely opinionated comment but I think its clearer if we stick to method names from the smart app executor base and not redefine them in a smart app.